core/src/main/java/de/danoeh/antennapod/core/storage/PodDBAdapter.java

```diff
@@ -1110,14 +1110,19 @@ public final Cursor getFeedItemCursor(final String[] ids) {
         return db.rawQuery(query, null);
     }
 
-    public final Cursor getFeedItemCursor(final String podcastUrl, final String episodeUrl) {
-        String escapedPodcastUrl = DatabaseUtils.sqlEscapeString(podcastUrl);
+    public final Cursor getFeedItemCursor(final String guid, final String episodeUrl) {
         String escapedEpisodeUrl = DatabaseUtils.sqlEscapeString(episodeUrl);
+        String whereClauseCondition = TABLE_NAME_FEED_MEDIA + "." + KEY_DOWNLOAD_URL + "=" + escapedEpisodeUrl;
+
+        if (guid != null) {
```

Comment:
I think the guid should also not be used when it is just an empty string or some whitespaces. Otherwise, this could affect unrelated episodes when publishers don't use guids correctly.

Reply:
What kind of assertion must this value pass to be considered a valid guid? Or just not empty string of white spaces?

Reply:
I added validation as proposed by you